src/sql/parts/notebook/notebook.component.ts

```diff
@@ -64,24 +64,32 @@ export class NotebookComponent extends AngularDisposable implements OnInit {
 		@Inject(IConnectionManagementService) private connectionManagementService: IConnectionManagementService,
 		@Inject(INotificationService) private notificationService: INotificationService,
 		@Inject(INotebookService) private notebookService: INotebookService,
-		@Inject(IBootstrapParams) private notebookParams: INotebookParams,
+		@Inject(IBootstrapParams) private _notebookParams: INotebookParams,
 		@Inject(IInstantiationService) private instantiationService: IInstantiationService,
 		@Inject(IContextMenuService) private contextMenuService: IContextMenuService,
 		@Inject(IContextViewService) private contextViewService: IContextViewService,
-		@Inject(IConnectionDialogService) private connectionDialogService: IConnectionDialogService
+		@Inject(IConnectionDialogService) private connectionDialogService: IConnectionDialogService,
+		@Inject(IEditorService) private editorService: IEditorService
 	) {
 		super();
-		this.profile = this.notebookParams!.profile;
+		this.profile = this._notebookParams!.profile;
 		this.isLoading = true;
 	}
 
 	ngOnInit() {
 		this._register(this.themeService.onDidColorThemeChange(this.updateTheme, this));
 		this.updateTheme(this.themeService.getColorTheme());
+		this.notebookService.addNotebookEditor(this);
```

Comment:
Now that we have editor registration at the component level, can we use it for all the active editor actions(Make dirty etc)?

Reply:
Yes this enables keyboard accelerators since we can query active editor, is that what you mean? For making dirty, I'm not sure on the blocker but I'm fairly sure you can plumb it through based on my changes, particularly the fact you have a reference to the input model in the notebook params, so that lets you call setDirty() on that.